### PR TITLE
cpu/esp32: fixes the memset optimization problem in esp_wifi/wpa_supplicant

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR adds a memset function `system_secure_memset` which cannot be optimized out by the compiler. It uses the libsodium approach of weak symbols. Function `system_secure_memset` calls the standard `memset` function. Calling an empty function declared with weak attribute after the `memset` call, prevents the compiler to optimize it out. The overhead is only one function call.

### Testing procedure

- For the time being only the successful compilation.
- Once PR #10762 is merged, `examples/gnrc_networking` with `esp_wif` netdev can be used to test.
   ```
   CFLAGS='-DESP_WIFI_SSID=\"<your SSID>\" -DESP_WIFI_PASS=\"your passphrase\"' USEMODULE="esp_wifi" make BOARD=esp32-wroom-32 -C examples/gnrc_networking flash
   ```

### Issues/PRs references

Fixes the problem described in #10751 for ESP32 vendor code.

### CODE REVIEW

File: cpu/esp32/syscalls.c [hunk 1]
```diff
@@ -595,3 +595,18 @@ void system_wdt_start (void)
...
+{
+    (void) s;
+    (void) c;
+    (void) n;
+}
+
+void *system_secure_memset(void *s, int c, size_t n)
+{
+    memset(s, c, n);
+    _system_prevent_memset_lto(s, c, n);
```
Comment (keestux) [inline cpu/esp32/syscalls.c]: In libsodium there is also
```
__asm__ __volatile__ ("" : : "r"(s) : "memory");
```
Did you leave that out on purpose?

Comment (gschorcht) [reply cpu/esp32/syscalls.c]: It's not necessary. The code generated is exactly the same.

w/o `__asm__ __volatile__ ("" : : "r"(s) : "memory");`
```
4008b210 <system_secure_memset>:
4008b210:	004136        	entry	a1, 32
4008b213:	20c440        	or	a12, a4, a4
4008b216:	20b330        	or	a11, a3, a3
4008b219:	20a220        	or	a10, a2, a2
4008b21c:	0464e5        	call8	4008f86c <memset>
4008b21f:	04cd      	mov.n	a12, a4
4008b221:	03bd      	mov.n	a11, a3
4008b223:	02ad      	mov.n	a10, a2
4008b225:	fffe25        	call8	4008b208 <_system_prevent_memset_lto>
4008b228:	f01d      	retw.n
```

with `__asm__ __volatile__ ("" : : "r"(s) : "memory");`
```
4008b210 <system_secure_memset>:
4008b210:	004136        	entry	a1, 32
4008b213:	20c440        	or	a12, a4, a4
4008b216:	20b330        	or	a11, a3, a3
4008b219:	20a220        	or	a10, a2, a2
4008b21c:	0464e5        	call8	4008f86c <memset>
4008b21f:	04cd      	mov.n	a12, a4
4008b221:	03bd      	mov.n	a11, a3
4008b223:	02ad      	mov.n	a10, a2
4008b225:	fffe25        	call8	4008b208 <_system_prevent_memset_lto>
4008b228:	f01d      	retw.n
```

Comment (keestux) [reply cpu/esp32/syscalls.c]: OK, thanks for checking.
ACK for the change then

**Comment by keestux** [comment]
@miri64 :-) I wasn't aware of the approve button, but now I am.

**Comment by miri64** [comment]
Still needs a Proxy-ACK, since you don't have write access to the RIOT repo yet.

**Pull request merged**